### PR TITLE
Update cluster autoscaler addon documentation for newly added flags

### PR DESCRIPTION
**Pull request by archups**
This PR is related to https://github.com/kubermatic/kubeone/pull/3305 and https://github.com/kubermatic/kubeone/pull/3306 changes for KubeOne. 
Also added documentation for the customization of KKP Autoscaler addon.

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold
This change is not available in 1.8 yet

File: content/kubeone/main/guides/autoscaler-addon/_index.en.md [hunk 1]
```diff
@@ -46,7 +46,43 @@ addons:
...
+  kubernetes: '1.29.4'   ## kubernetes version
+cloudProvider:  ## This field is sourced automatically if terraform is used for the cluster
+  aws: {}
+addons:
+  enable: true
+  addons:
+  - name: cluster-autoscaler
+    params:
+      CLUSTER_AUTOSCALER_BALANCE_SIMILAR_NODE_GROUP: "true"
+```
```
Comment (xmudrii) [inline content/kubeone/main/guides/autoscaler-addon/_index.en.md:83]: Can we put this in the step 1 instead, in an information or a warning block? This step is more about modifying the cluster-autoscaler itself (i.e. YAML manifests), but this is more about additionally configuring the cluster-autoscaler addon.

File: content/kubeone/main/guides/autoscaler-addon/_index.en.md [hunk 2]
```diff
@@ -212,3 +248,5 @@ That is it! You have successfully deployed Kubernetes autoscaler on the KubeOne
 [embedded-addons]: {{< ref "../../guides/addons/#overriding-embedded-eddons" >}}
 [ca-faq]: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md
 [ca-faq-what-is]: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-is-cluster-autoscaler
+[enforce-node-group-min-size]: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#my-cluster-is-below-minimum--above-maximum-number-of-nodes-but-ca-did-not-fix-that-why
+[balance-similar-node-groups]: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#im-running-cluster-with-nodes-in-multiple-zones-for-ha-purposes-is-that-supported-by-cluster-autoscaler
```
Comment (xmudrii) [inline content/kubeone/main/guides/autoscaler-addon/_index.en.md:252]: Can you instead use permalinks here?

File: content/kubeone/main/guides/autoscaler-addon/_index.en.md [hunk 3]
```diff
@@ -46,7 +46,43 @@ addons:
 
 If you wish to change some of the properties, such as timeout for scaling up/down, you’ll need to provide the appropriate command-line flags to cluster-autoscaler. For that, you’ll need to override the cluster-autoscaler addon embedded in the KubeOne binary with your addon.
 
-To find out how to override embedded add-ons, please check the [Addons document][embedded-addons]. For more information regarding available configuration parameters and options, please check the [Cluster Autoscaler FAQ][ca-faq].
+To find out how to override embedded add-ons, please check the [Addons document][embedded-addons].
+
+If you are facing issue similar to the one described over [here][enforce-node-group-min-size] in the [Cluster Autoscaler FAQ][ca-faq], pass down the param `CLUSTER_AUTOSCALER_ENFORCE_NODE_GROUP_MIN_SIZE` to the cluster autoscaler addon as shown below:
```
Comment (xmudrii) [inline content/kubeone/main/guides/autoscaler-addon/_index.en.md:51]:
```suggestion
If you're facing an issue where your [cluster is below the minimum or above the maximum number of nodes, but the cluster-autoscaler is not fixing it][ca-faq], you need to pass down the `CLUSTER_AUTOSCALER_ENFORCE_NODE_GROUP_MIN_SIZE` parameter to the cluster-autoscaler addon as shown below:
```

File: content/kubeone/main/guides/autoscaler-addon/_index.en.md [hunk 4]
```diff
@@ -46,7 +46,43 @@ addons:
...
+  aws: {}
+addons:
+  enable: true
+  addons:
+  - name: cluster-autoscaler
+    params:
+      CLUSTER_AUTOSCALER_ENFORCE_NODE_GROUP_MIN_SIZE: "true"
+```
+
+If you are facing issue similar to the one described over [here][balance-similar-node-groups] in the [Cluster Autoscaler FAQ][ca-faq], pass down the param `CLUSTER_AUTOSCALER_BALANCE_SIMILAR_NODE_GROUP` to the cluster autoscaler addon as shown below:
```
Comment (xmudrii) [inline content/kubeone/main/guides/autoscaler-addon/_index.en.md:68]:
```suggestion
If you're running a cluster with nodes in the multiple zones for the HA purposes, consider setting the `CLUSTER_AUTOSCALER_BALANCE_SIMILAR_NODE_GROUP` parameter to `true` as shown below. For more information about this parameter, check the [Cluster Autoscaler FAQ][balance-similar-node-groups].
```

Comment (xmudrii) [reply content/kubeone/main/guides/autoscaler-addon/_index.en.md:68]: Additionally, to make it easier to follow the document, maybe we should provide just one YAML example with both arguments set, eventually with a comment that you can remove the parameter that you don't need.

File: content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md [hunk 5]
```diff
@@ -187,7 +187,16 @@ You can delete autoscaler from where you edit it above and select delete.
 ![Delete Autoscaler](/img/kubermatic/v2.25/tutorials/cluster-autoscaler/delete-autoscaler.png?classes=shadow,border "Delete Autoscaler")
 
 
- Once it has been deleted, you can check the cluster to ensure that the autoscaler has been deleted using the command `kubectl get pods -n kube-system`.
+ Once it has been deleted, you can check the cluster to ensure that the cluster autoscaler has been deleted using the command `kubectl get pods -n kube-system`.
+
+
+## Customize KKP Autoscaler
```
Comment (xmudrii) [inline content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md:193]: Same as for the KubeOne document, I'd recommend using permalinks here.

File: content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md [hunk 6]
```diff
@@ -187,7 +187,16 @@ You can delete autoscaler from where you edit it above and select delete.
...
+
+You can customize the cluster autoscaler addon in order to override the cluster autoscaler deployment definition to set or pass the required flag(s) by following the instructions provided over [here](https://docs.kubermatic.com/kubermatic/v2.25/architecture/concept/kkp-concepts/addons/#custom-addons), if you are facing any of the below mentioned issues as described in the [cluster autoscaler FAQs](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md):
```
Comment (xmudrii) [inline content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md:195]: What about something like this:
```suggestion
You can customize the cluster autoscaler addon in order to override the cluster autoscaler deployment definition to set or pass the required flag(s) by following the instructions provided [in the Addons document](https://docs.kubermatic.com/kubermatic/v2.25/architecture/concept/kkp-concepts/addons/#custom-addons).

### Suggestions and Known Issues

Some of [the know issues in the cluster-autoscaler](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md) can be mitigated by adjusting the cluster-autoscaler deployment to pass additional flags, for example:
```

Comment (xmudrii) [reply content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md:195]: Also please use `ref` to access other KKP documents, such as `{{< ref "../../architecture/concept/kkp-concepts/addons/#custom-addons" >}}` (I didn't test this URL, please make sure to test it and fix it if needed).

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md (hunk 5 above)
Comment (archups) [reply content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md:193]: fixed!

File: content/kubeone/main/guides/autoscaler-addon/_index.en.md (hunk 1 above)
Comment (archups) [reply content/kubeone/main/guides/autoscaler-addon/_index.en.md]: fixed!

File: content/kubeone/main/guides/autoscaler-addon/_index.en.md (hunk 2 above)
Comment (archups) [reply content/kubeone/main/guides/autoscaler-addon/_index.en.md]: done!

File: content/kubeone/main/guides/autoscaler-addon/_index.en.md (hunk 3 above)
Comment (archups) [reply content/kubeone/main/guides/autoscaler-addon/_index.en.md]: done!

File: content/kubeone/main/guides/autoscaler-addon/_index.en.md (hunk 4 above)
Comment (archups) [reply content/kubeone/main/guides/autoscaler-addon/_index.en.md]: done!

File: content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md (hunk 6 above)
Comment (archups) [reply content/kubermatic/v2.25/tutorials-howtos/kkp-autoscaler/_index.en.md]: done!

**Comment by xmudrii** [comment]
KubeOne part LGTM, @embik do you want to take care of the KKP part?
/lgtm

**Review by embik — APPROVED** [review]
/approve
/hold cancel

**Pull request merged**